### PR TITLE
feat(heatmap): Sync cursor from xy charts to Heatmap

### PR DESCRIPTION
**Pull request by qn895**
## Summary

Cursor positions from xy charts are now synced with the Heatmap.

https://user-images.githubusercontent.com/43350163/174882861-3c86740b-c000-4562-8ec8-e0b298b3bcd6.mov

## Details

This PR is a follow up of https://github.com/elastic/elastic-charts/pull/1668. It converts the continuous external pointer events from xy charts into visible bands on the heatmap. It also highlights the grid cell when cursor is hovered.

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [ ] The `:theme` label has been added and the `@elastic/eui-design` team has been pinged when there are `Theme` API changes
- [x] All related issues have been linked (i.e. `closes #123`, `fixes #123`)
- [ ] New public API exports have been added to `packages/charts/src/index.ts`
- [ ] Unit tests have been added or updated to match the most common scenarios
- [x] The proper documentation and/or storybook story has been added or updated
- [x] The code has been checked for cross-browser compatibility (Chrome, Firefox, Safari, Edge)
- [ ] Visual changes have been tested with all available themes including `dark`, `light`, `eui-dark` & `eui-light`

### CODE REVIEW

File: storybook/stories/interactions/19_multi_chart_cursor_sync.story.tsx [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    .reduce<{ x: number; y: string; value: number }[]>((acc, [x, y], i) => {
+      if (i % 5 === 0) {
+        acc.push({ x, y: '2fd4e', value: y });
+      } else {
+        acc[acc.length - 1].value += y;
+      }
+
+      return acc;
+    }, [])
+    .map(({ x, y, value }, i) => (rng()) > 0.6 ? { x, y, value: null } : { x, y, value })),
```
Comment (nickofthyme) [inline storybook/stories/interactions/19_multi_chart_cursor_sync.story.tsx:42]:
```suggestion
    .map(({ x, y, value }, i) => (rng()) > 0.6 ? { x, y, value: null } : { x, y, value }))
```

File: packages/charts/src/chart_types/heatmap/layout/types/viewmodel_types.ts [hunk 2]
```diff
@@ -99,6 +99,9 @@ export type PickHighlightedArea = (
   y: Array<NonNullable<PrimitiveValue>>,
 ) => Rect | null;
```
Comment (nickofthyme) [inline packages/charts/src/chart_types/heatmap/layout/types/viewmodel_types.ts:101]: All code changes LGTM. I did find one edge case when playing around with it.

When you hover over the heatmap on the last _bucket_ on the right, both xy charts correctly render the cursor. But when you hover over either of the xy charts only the second to last bucket on the heatmap will be highlighted.

Screen Recording 2022-07-29 at 09 06 01 AM

> The works correctly on the first bucket on the left end of the domain.

Comment (markov00) [reply packages/charts/src/chart_types/heatmap/layout/types/viewmodel_types.ts:101]: Fixed in https://github.com/elastic/elastic-charts/pull/1721/commits/56799b4d097926db47aa27aeeb0f7f18c4380dfd

**Comment by markov00** [comment]
buildkite update screenshots

**Comment by markov00** [comment]
buildkite update screenshots

**Review by nickofthyme — APPROVED** [review]

**Pull request merged**